app/services/amr/analytics_unvalidated_meter_collection_factory.rb

```diff
@@ -73,7 +73,7 @@ def add_amr_data(active_record_meter)
 
       hash_of_date_formats = AmrDataFeedConfig.pluck(:id, :date_format).to_h
 
-      AmrDataFeedReading.where(meter_id: active_record_meter.id).each do |reading|
+      AmrDataFeedReading.order(created_at: :asc).where(meter_id: active_record_meter.id).each do |reading|
```

Comment:
Forcing this rather than relying on chance means that when we have multiple records for the same day (as we now do) that we pick the *last* created. These get put in a hash keyed on date, so the latest one put in the hash is the one used.